pkg/watermark/generic/jetstream/generic.go

```diff
@@ -51,25 +51,71 @@ func BuildWatermarkProgressors(ctx context.Context, vertexInstance *v1alpha1.Ver
 	}
 
 	pipelineName := vertexInstance.Vertex.Spec.PipelineName
-	fromBucket := vertexInstance.Vertex.GetFromBuckets()[0]
 
+	fetchWatermark, err := buildFetcher(ctx, vertexInstance)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	publishWatermark, err := buildPublishers(ctx, pipelineName, vertexInstance)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return fetchWatermark, publishWatermark, nil
+}
+
+func buildFetcher(ctx context.Context, vertexInstance *v1alpha1.VertexInstance) (fetch.Fetcher, error) {
```

Comment:
Can you add a comment for this function?

Reply:
sure, can add comments for the various functions I added here